openreferee_server/operations.py

```diff
@@ -87,3 +94,32 @@ def cleanup_event(event):
     session = setup_requests_session(event.token)
     cleanup_event_tags(session, event)
     cleanup_file_types(session, event)
+
+
+def process_editable_files(session, files, endpoints):
+    uploaded = defaultdict(list)
+    for _file in files:
+        if _file['filename'].split(".")[-1] == 'pdf':
+            upload = process_pdf(_file['external_download_url'], session, endpoints['file_upload'])
+            uploaded[_file['file_type']].append(upload['uuid'])
+    session.post(endpoints['revisions']['replace'], json={
+        'files': uploaded,
+        'state': 'ready_for_review'
+    })
+
+
+def process_pdf(url, session, upload_endpoint):
+    pdf_writer = PdfFileWriter()
+    rf = urlopen(Request(url, headers=session.headers))
+    pdf_reader = PdfFileReader(io.BytesIO(rf.read()))
+    watermark_pdf = PdfFileReader(open(Path(__file__).parent / 'watermark.pdf', 'rb'))
+    watermark_page = watermark_pdf.getPage(0)
+    for page in range(0, pdf_reader.numPages):
+        page = pdf_reader.getPage(0)
+        page.mergePage(watermark_page)
+        pdf_writer.addPage(page)
+    with tempfile.NamedTemporaryFile(suffix='.pdf') as f:
```

Comment:
I don't think you need to write it to disk even temporarily. You can pass a file-like object (e.g. `BytesIO`) in requests' `files` dict, and most likely the pdf writer also allows you to pass a filelike object.

Reply:
It works well, I initially used a tempfile to preserve the extension but seems I can just pass it along.

Reply:
Yes, you can pass a tuple in the `files` dict containing details about the file(-object)